examples/pipelines/controlnet-interior-design/components/retrieve_images/src/main.py

```diff
@@ -0,0 +1,89 @@
+"""
+This component retrieves image URLs from LAION-5B based on a set of seed prompts.
+"""
+import logging
+from typing import List
+
+import dask.dataframe as dd
+import dask.array as da
+
+from clip_client import ClipClient, Modality
+
+from fondant.component import FondantTransformComponent
+from fondant.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger(__name__)
+
+
+def query_clip_client(text: str, client: ClipClient) -> List[str]:
+    """
+    Given a text query and a ClipClient instance, this function retrieves
+    image URLs related to the query.
+
+    Args:
+        text: the text query
+        client (ClipClient): an instance of ClipClient used to query the images
+
+    Returns:
+        results: a list of strings, each representing a URL of an image related to the query
+    """
+    results = client.query(text=text)
+    results = [i["url"] for i in results]
+
+    return results
+
+
+class RetrieveImagesComponent(FondantTransformComponent):
+    """
+    Component that retrieves image URLs from LAION-5B based on a set of prompts.
+    """
+
+    def transform(
+        self,
+        dataframe: dd.DataFrame,
+        *,
+        num_images: int,
+        aesthetic_score: int,
+        aesthetic_weight: float
+    ) -> dd.DataFrame:
+        """
+        Args:
+            dataframe: Dask dataframe
+            num_images: number of images to retrieve for each prompt
+            aesthetic_score: ranking score for aesthetic, higher is prettier.
+            aesthetic_weight: weight of the aesthetic score, between 0 and 1.
+
+        Returns:
+            Dask dataframe
+        """
+        client = ClipClient(
+            url="https://knn.laion.ai/knn-service",
+            indice_name="laion5B-L-14",
+            num_images=num_images,
+            aesthetic_score=aesthetic_score,
+            aesthetic_weight=aesthetic_weight,
+            modality=Modality.IMAGE,
+        )
+
+        dataframe["images_url"] = dataframe["prompts_text"].map_partitions(
+            lambda example: example.apply(query_clip_client, args=(client,)),
+            meta=("images_url", "str"),
+        )
+
+        # unpack list of urls
+        dataframe = dataframe.explode("images_url")
+
+        # add index
+        index_list = [idx for idx in range(len(dataframe))]
```

Comment:
For this one you can do as follows: 
```
dask_df['id'] = dask_df.assign(id=1).row_num.cumsum()
```

Here, we take a temporary column and assign it a value of 1, then we use `cumsum()` to calculative the cumulative sum of all previous values (emulating your `range(len)`). 

You will get the same results but without any array/list in memory. 

I don't expect users to figure this one right off the bat so either we make it very explicit with documentation or we handle this internally

Reply:
what I meant by emulating `range(len)` is that you can set the index incrementally with `cumsum()` but without loading anything in memory. There is some computation that happens but should be lightweight

Reply:
Yes, your solution is even better, just as a side note :slightly_smiling_face: 

But shouldn't we actually leverage the LAION ids here?

Reply:
@PhilippeMoussalli this gives me: `AttributeError: 'DataFrame' object has no attribute 'row_num'`

Reply:
sorry it should be `id` instead of `row_num`

```
dask_df['id'] = dask_df.assign(id=1).id.cumsum()

```